### ISSUE TITLE
Convert bash script js into GULP bakery tasks: isolate each task of the Bakery CLI

### ISSUE THREAD

**Issue by brittweinstein** [openstax/cnx-recipes#1227]
Tasks List:
- [ ] Fetch
- [ ] Assemble
- [ ] GetRecipe (latest)
- [ ] Bake
- [ ] Mathify
- [ ] GetStyle (latest)
- [ ] InjectStyle
- [ ] BuildPDF
- [ ] DiffRecipe (possibly out of scope)
- [ ] DiffBook
When a task is run:
- [ ] It verifies the artifacts from each previous step exists
- [ ] If an artifacts exist, and we don’t use a ‘force’ option, we are done
- [ ] If an artifact doesn’t exist, run the step
Stop once we get to the specified step
- [ ] User must have the option to force none, one, or any given amount of steps before the specified one.

Notes: Rationale for dropping Gulp:
One fewer global dependency
Gulp would hide errors and silently fail
Gulp would print out a bunch of garbage when tasks run in parallel
Gulp’s intended use case does not match ours exceptionally well, although it is a task runner
Gulp and yargs fight about getting to control the help command
Niche JS libraries do better in their areas than Gulp - Alternatives:
Yargs: helps us create a simple and more customizable CLI than gulp does, good support for --help (list options and commands) as well as default values
Async: does the same with async/callback based control flow, but a more comprehensive library and more thorough documentation
